docs/promise-chaining.md

```diff
@@ -0,0 +1,90 @@
+# Promise Chaining
+
+Imagine that we have a sequence of asynchronous tasks that we would love to be done one after another, while passing the result through the promise chain. This can be very useful for something like loading scripts or other resources. Promises provide a couple of simple ways to achieve that.
+
+Lets take the code we executed from the JS file and analyze it step by step:
+
+    new Promise(function(resolve, reject) { 
+    setTimeout(() => resolve(1), 1000);
+Promise is set to a timeout of 1000 ms, which is 1 second.
+
+    }).then(function(res) {
+    alert("Value: " + res);
+    return res * 2;
+Handler `.then` is called taking the promise result `res = 1` as an argument. Result of our initial promise is `1`, so this code alerts `Value: 1` and returns `2` to be passed to the next handler.
+
+    }).then(function(res) {
+    alert("Value: " + res);
+    return res * 2;
+We have another `.then` handler, which is taking the previous handlers result, which is `2`. This code alerts `Value: 2`. Again, the result is returned as the result of the promise and being multiplied by 2, so in the next handler, the `res` argument equals `4`.
+
+    }).then(function(res) {
```

Comment:
```
}).then(function(res) {
    alert("Value: " + res);
    return res * 2;
```

In this part the }) from the .md as it it looking like without work you have made the .md